src/company_gui.cpp

```diff
@@ -1827,267 +1845,217 @@ struct CompanyInfrastructureWindow : Window
-				max_cost = std::max(max_cost, SignalMaintenanceCost(c->infrastructure.signal));
-				uint32_t road_total = c->infrastructure.GetRoadTotal();
-				uint32_t tram_total = c->infrastructure.GetTramTotal();
-				for (RoadType rt = ROADTYPE_BEGIN; rt < ROADTYPE_END; rt++) {
-					max_val = std::max(max_val, c->infrastructure.road[rt]);
-					max_cost = std::max(max_cost, RoadMaintenanceCost(rt, c->infrastructure.road[rt], RoadTypeIsRoad(rt) ? road_total : tram_total));
+		infrastructurelist.emplace_back(CIW_ENTRY_LABEL_VALUES, STR_COMPANY_INFRASTRUCTURE_VIEW_CANALS, c->infrastructure.water, maintenance_cost);
 
-				}
-				max_val = std::max(max_val, c->infrastructure.water);
-				max_cost = std::max(max_cost, CanalMaintenanceCost(c->infrastructure.water));
-				max_val = std::max(max_val, c->infrastructure.station);
-				max_cost = std::max(max_cost, StationMaintenanceCost(c->infrastructure.station));
-				max_val = std::max(max_val, c->infrastructure.airport);
-				max_cost = std::max(max_cost, AirportMaintenanceCost(c->index));
-
-				SetDParamMaxValue(0, max_val);
-				uint count_width = GetStringBoundingBox(STR_JUST_COMMA).width + WidgetDimensions::scaled.hsep_indent; // Reserve some wiggle room
-
-				if (_settings_game.economy.infrastructure_maintenance) {
-					StringID str_total = TimerGameEconomy::UsingWallclockUnits() ? STR_COMPANY_INFRASTRUCTURE_VIEW_TOTAL_PERIOD : STR_COMPANY_INFRASTRUCTURE_VIEW_TOTAL_YEAR;
-					SetDParamMaxValue(0, this->GetTotalMaintenanceCost() * 12); // Convert to per year
-					this->total_width = GetStringBoundingBox(str_total).width + WidgetDimensions::scaled.hsep_indent * 2;
-					size.width = std::max(size.width, this->total_width);
-
-					SetDParamMaxValue(0, max_cost * 12); // Convert to per year
-					count_width += std::max(this->total_width, GetStringBoundingBox(str_total).width);
-				}
 
-				size.width = std::max(size.width, count_width);
+		/* air-ways. */
+		infrastructurelist.emplace_back(CIW_ENTRY_HEADER, STR_COMPANY_INFRASTRUCTURE_VIEW_STATION_SECT);
 
-				/* Set height of the total line. */
-				if (widget == WID_CI_TOTAL) {
-					size.height = _settings_game.economy.infrastructure_maintenance ? std::max<uint>(size.height, WidgetDimensions::scaled.vsep_normal + GetCharacterHeight(FS_NORMAL)) : 0;
-				}
+		maintenance_cost        = StationMaintenanceCost(c->infrastructure.station);
+		total_maintenance_cost += maintenance_cost;
+
+		infrastructurelist.emplace_back(CIW_ENTRY_LABEL_VALUES, STR_COMPANY_INFRASTRUCTURE_VIEW_STATIONS, c->infrastructure.station, maintenance_cost);
+
+		maintenance_cost        = AirportMaintenanceCost(c->index);
+		total_maintenance_cost += maintenance_cost;
+
+		infrastructurelist.emplace_back(CIW_ENTRY_LABEL_VALUES, STR_COMPANY_INFRASTRUCTURE_VIEW_AIRPORTS, c->infrastructure.airport, maintenance_cost);
+
+
+		/* total maintenance cost. */
+		infrastructurelist.emplace_back(CIW_ENTRY_TOTAL_COST, STR_COMPANY_VIEW_INFRASTRUCTURE_NONE, 0, total_maintenance_cost);
```

Comment:
It's difficult to see from the diff, but I think all the extra blank lines makes this harder to read.

Laid out like as below would give more logical grouping to each section, in my opinion -- each value line is more closely associated with its value calculation. The same applies to the early sections too.

```c++
		/* water-ways. */
		infrastructurelist.emplace_back(CIW_ENTRY_HEADER, STR_COMPANY_INFRASTRUCTURE_VIEW_WATER_SECT);

		maintenance_cost        = CanalMaintenanceCost(c->infrastructure.water);
		total_maintenance_cost += maintenance_cost;
		infrastructurelist.emplace_back(CIW_ENTRY_LABEL_VALUES, STR_COMPANY_INFRASTRUCTURE_VIEW_CANALS, c->infrastructure.water, maintenance_cost);

		/* air-ways. */
		infrastructurelist.emplace_back(CIW_ENTRY_HEADER, STR_COMPANY_INFRASTRUCTURE_VIEW_STATION_SECT);

		maintenance_cost        = StationMaintenanceCost(c->infrastructure.station);
		total_maintenance_cost += maintenance_cost;
		infrastructurelist.emplace_back(CIW_ENTRY_LABEL_VALUES, STR_COMPANY_INFRASTRUCTURE_VIEW_STATIONS, c->infrastructure.station, maintenance_cost);

		maintenance_cost        = AirportMaintenanceCost(c->index);
		total_maintenance_cost += maintenance_cost;
		infrastructurelist.emplace_back(CIW_ENTRY_LABEL_VALUES, STR_COMPANY_INFRASTRUCTURE_VIEW_AIRPORTS, c->infrastructure.airport, maintenance_cost);

		/* total maintenance cost. */
		infrastructurelist.emplace_back(CIW_ENTRY_TOTAL_COST, STR_COMPANY_VIEW_INFRASTRUCTURE_NONE, 0, total_maintenance_cost);
```